EIPS/eip-5727.md

```diff
@@ -0,0 +1,596 @@
+     * @param value The value of the token minted
+     */
+    event Minted(address indexed soul, uint256 indexed tokenId, uint256 value);
+
+    /**
+     * @dev MUST emit when a token is revoked.
+     * @param soul The owner soul of the revoked token
+     * @param tokenId The revoked token
+     */
+    event Revoked(address indexed soul, uint256 indexed tokenId);
+
+    /**
+     * @dev MUST emit when a token is charged.
+     * @param tokenId The token to charge
+     * @param value The value to charge
+     */
+    event Charged(uint256 indexed tokenId, uint256 value);
+
+    /**
+     * @dev MUST emit when a token is consumed.
+     * @param tokenId The token to consume
+     * @param value The value to consume
+     */
+    event Consumed(uint256 indexed tokenId, uint256 value);
+
+    /**
+     * @dev MUST emit when a token is destroyed.
+     * @param soul The owner soul of the destroyed token
+     * @param tokenId The token to destroy.
+     */
+    event Destroyed(address indexed soul, uint256 indexed tokenId);
+
+    /**
+     * @dev MUST emit when the slot of a token is set or changed.
+     * @param tokenId The token of which slot is set or changed
+     * @param oldSlot The previous slot of the token
+     * @param newSlot The updated slot of the token
+     */
+    event SlotChanged(
+        uint256 indexed tokenId,
+        uint256 indexed oldSlot,
+        uint256 indexed newSlot
+    );
+
+    /**
+     * @notice Get the value of a token.
+     * @dev MUST revert if the `tokenId` does not exist
+     * @param tokenId the token for which to query the balance
+     * @return The value of `tokenId`
+     */
+    function valueOf(uint256 tokenId) external view returns (uint256);
+
+    /**
+     * @notice Get the slot of a token.
+     * @dev MUST revert if the `tokenId` does not exist
+     * @param tokenId the token for which to query the slot
+     * @return The slot of `tokenId`
+     */
+    function slotOf(uint256 tokenId) external view returns (uint256);
+
+    /**
+     * @notice Get the owner soul of a token.
+     * @dev MUST revert if the `tokenId` does not exist
+     * @param tokenId the token for which to query the owner soul
+     * @return The address of the owner soul of `tokenId`
+     */
+    function soulOf(uint256 tokenId) external view returns (address);
+
+    /**
+     * @notice Get the validity of a token.
+     * @dev MUST revert if the `tokenId` does not exist
+     * @param tokenId the token for which to query the validity
+     * @return If the token is valid
+     */
+    function isValid(uint256 tokenId) external view returns (bool);
+
+    /**
+     * @notice Get the issuer of a token.
+     * @dev MUST revert if the `tokenId` does not exist
+     * @param tokenId the token for which to query the issuer
+     * @return The address of the issuer of `tokenId`
+     */
+    function issuerOf(uint256 tokenId) external view returns (address);
+}
+```
+
+### Extensions
+#### Enumerable
```

Comment:
Would be nice to see some explanation of the extensions, and who might want to implement them.